src/lib/middleware/services/config_cbor_map_service.cpp

```diff
@@ -0,0 +1,104 @@
+#include "config_cbor_map_service.h"
+#include "FreeRTOS.h"
+#include "bm_service.h"
+#include "bm_service_common.h"
+#include "config_cbor_map_srv_reply_msg.h"
+#include "config_cbor_map_srv_request_msg.h"
+#include "device_info.h"
+
+#define CONFIG_MAP_SUFFIX "/config_map"
+
+cfg::Configuration* _sys_config;
+cfg::Configuration* _hw_config;
+cfg::Configuration* _usr_config;
+
+static bool config_map_service_handler(size_t service_strlen, const char *service,
+                                   size_t req_data_len, uint8_t *req_data, size_t &buffer_len,
+                                   uint8_t *reply_data);
+
+void config_cbor_map_service_init(cfg::Configuration& hw_cfg, cfg::Configuration& sys_config, cfg::Configuration& user_config) {
+  _sys_config = &sys_config;
+  _hw_config = &hw_cfg;
+  _usr_config = &user_config;
+  static char config_map_str[BM_SERVICE_MAX_SERVICE_STRLEN];
+  size_t topic_strlen = snprintf(config_map_str, sizeof(config_map_str),
+                                 "%" PRIx64 "%s", getNodeId(), CONFIG_MAP_SUFFIX);
+  if (topic_strlen > 0) {
+    bm_service_register(topic_strlen, config_map_str, config_map_service_handler);
+  } else {
+    printf("Failed to register sys info service\n");
+  }
+}
+
+bool config_cbor_map_service_request(uint64_t target_node_id, uint32_t partition_id, bm_service_reply_cb reply_cb, uint32_t timeout_s) { 
+  bool rval = false;
+  char *target_service_str = static_cast<char *>(pvPortMalloc(BM_SERVICE_MAX_SERVICE_STRLEN));
+  configASSERT(target_service_str);
+  size_t target_service_strlen =
+      snprintf(target_service_str, BM_SERVICE_MAX_SERVICE_STRLEN, "%" PRIx64 "%s",
+               target_node_id, CONFIG_MAP_SUFFIX);
+  ConfigCborMapSrvRequestMsg::Data req = {partition_id};
+  static constexpr size_t cbor_buflen = sizeof(ConfigCborMapSrvRequestMsg::Data) + 25; // Abitrary padding.
```

Comment:
why do we need padding?

Reply:
The padding accounts for Cbor's encoding inefficiency. 25 as padding seems like a reasonable balance between "it-works", without over-allocating for this static length piece of data.

Reply:
Gotcha!